### DISCUSSION TITLE
[archived] "Loops Walk Through" docs discussion

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#9478] [General]
# argo-workflows/walk-through/loops/

https://argoproj.github.io/argo-workflows/walk-through/loops/

**Comment by hxlee309** [comment]
Hi, the last one doesn't work for me

```
Name:                loops-param-result-ncrhw
Namespace:           argo
ServiceAccount:      unset (will run with the default ServiceAccount)
Status:              Failed
Message:             child 'loops-param-result-ncrhw-1237519262' failed
Conditions:          
 PodRunning          False
 Completed           True
Created:             Wed Aug 31 01:15:32 -0700 (31 seconds ago)
Started:             Wed Aug 31 01:15:33 -0700 (30 seconds ago)
Finished:            Wed Aug 31 01:16:03 -0700 (now)
Duration:            30 seconds
Progress:            0/1
ResourcesDuration:   11s*(1 cpu),11s*(100Mi memory)

STEP                         TEMPLATE                   PODNAME                              DURATION  MESSAGE
 ✖ loops-param-result-ncrhw  loop-param-result-example                                                 child 'loops-param-result-ncrhw-1237519262' failed                                                                                                                                            
 └───⚠ generate              gen-number-list            loops-param-result-ncrhw-1237519262  15s       Error (exit code 1): pods "loops-param-result-ncrhw-1237519262" is forbidden: User "system:serviceaccount:argo:default" cannot patch resource "pods" in API group "" in the namespace "argo"  

This workflow does not have security context set. You can run your workflow pods more securely by setting it.
Learn more at https://argoproj.github.io/argo-workflows/workflow-pod-security-context/
```

**Reply by FeddeX99** [reply]
Hi, you might be able to solve this by running:
```sh
kubectl create rolebinding default-admin --clusterrole=admin --serviceaccount=argo:default --namespace=argo
```
It is not the safe and right option, but it could help you make it work. I faced a similar error message and this resolved the problem.

**Reply by xietianyu** [reply]
give serviceaccount 'default' in namespace 'argo'  a clusterrole
```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: role-namespace
rules:
  - apiGroups:
      - ""
    resources:
      - pod
    verbs:
      - get
      - list
      - watch
      - create
      - patch
      - update
      - delete

**Reply by agilgur5** [reply]
For the minimal RBAC needed for Workflow pods, see https://argo-workflows.readthedocs.io/en/latest/workflow-rbac/

**Comment by dazinator** [comment]
Is there a max limit for loop iterations? Could 100000 items be sent?

**Comment by dazinator** [comment]
Do loops run in parallel? Or sequential?

**Reply by dazinator** [reply]
Sorry from diagram it looks like in parallel.. what if you wanted a sequential loop?

**Reply by agilgur5** [reply]
You can use `parallelism` or other [synchronization](https://argo-workflows.readthedocs.io/en/latest/synchronization/) features to achieve sequential loop execution

**Comment by Lynggaard91** [comment]
You won't be able to `argo submit` the last example as it is a WorkflowTemplate and not a Workflow. Presuming your environment is based on the tutorial, create the WorkflowTemplate in the Namespace where Argo is running. You do this by `argo template create workflowtemplate-loops.yaml -n argo` and hereafter run the WorkflowTemplate by submitting it `argo submit --from workflowtemplate/loop-test --watch`
If you submit the WorkflowTemplate without the namespace flag (outside of the Argo namespace), you'll have to grant the SA of the namespace where you created the WorkflowTemplate access to the Argo namespace. Typically people mistakenly create the WorkflowTemplate in the Default namespace. FeddeX99 comment (top) shows how to do so.

**Comment by frankrefischer** [comment]
We try to run the example "withParam example from another step in the workflow" but we get the following error: 

```
Error (exit code 1):
WorkflowTaskResult.argoproj.io "loops-param-result-8x7dh-968208049" is invalid:
metadata.ownerReferences.uid: Invalid value: "": uid must not be empty
```
Why? We used the installation steps from Quick Start with Version 3.4.13
